massgov/mayflower-patternlab
# MGRP-258 document mayflower atoms

This PR adds documentation to Mayflower atoms with the exception of icon pattern variations (see discussion below).   

Addresses ticket: https://jira.state.ma.us/browse/MGRP-258

### Acceptance notes
Each component should have a file named "name-of-template".md where the name of template is the same as the twig and json file.
The document should set the development status for each component of: depreciated, alpha, beta or stable.
The document should follow the example pattern below
Update the readme to include information about documentation.

The documentation is in the form of `.md` files located within the same directory as the pattern which they are documenting.

### Example .md template:

> ---
> title: Button main
> ---
> Description: an atom for a link styled as a button.
> ## State: ALPHA
> ###  Notes
> Describe the roadmap to get the component to a stable state. And any other notes for rendering.
> ### Used in: 
> List of atoms/molecules/organisms used in. 
> ### Contains
> List of atoms/molecules/organisms that are used.
> ### Variables 
> ~~~ 
> buttonMain {
>     href:
>         type: string/required
>     text: 
>         type: string/required
> }
> ~~~

### To Test:
1. Pull the dev branch from this repo (PR has been merged)
2. From within `styleguide/`, run `gulp` to build Mayflower
3. Visit localhost:3000
4. Use the top blue navigation menu to browse through the list of `ATOMS`, note patterns which have DEPRECATED status are indicated in menu (gif below)
5. Click any atom pattern, show the pattern info (under cog icon in top right corner of nav) (see gif below)
6. Bottom half of your window should change to show pattern information with twig/html tabs on the right, and documentation on the left.  This left pane is populated via the `.md` files which live in the codebase right next to (and named the same as) the twig templates which they describe. (see gif below)

[![https://gyazo.com/b5fcf0ada7fe1bb6a1a4049b1d61ac42](https://i.gyazo.com/b5fcf0ada7fe1bb6a1a4049b1d61ac42.gif)](https://gyazo.com/b5fcf0ada7fe1bb6a1a4049b1d61ac42)
### Notes:

I think it would be good for someone to revisit the development state that I used for each of these patterns.  I labelled most alpha with the exception of maybe one or two that could be deprecated.
